components/Map/index.js

```diff
@@ -0,0 +1,62 @@
+import { MapContainer, TileLayer, Marker, useMap } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
+import styled from "styled-components";
+import { Icon } from "leaflet";
+import { useEffect } from "react";
+
+const UpdateMapView = ({ lat, lng }) => {
+  const map = useMap();
+
+  useEffect(() => {
+    if (lat && lng) {
+      map.setView([lat, lng], 13);
+    }
+  }, [lat, lng, map]);
+
+  return null;
+};
```

Comment:
This looks a bit strange to me. I see a function component that does not return any JSX but instead makes ose of a useEffect? Is this from any leaflet documentation? I am not that familiar with leaflet but I'm interested where this code comes from. Maybe we can talk about it.